packages/vibrant-components/src/lib/Table/Table.tsx

```diff
@@ -88,6 +92,124 @@ export const Table = <Data extends Record<string, any>, RowKey extends keyof Dat
   const [selectedCellKey, setSelectedCellKey] = useState<string>();
   const isCellClickEnabled = columns?.some(column => isDefined(column.onDataCell));
 
+  const [selectedRange, setSelectedRange] = useState<TableCellRange>();
+  const [isSelectingRange, setIsSelectingRange] = useState(false);
+
+  const isCellInSelectedRange = (rowIdx: number, colIdx: number) => {
+    if (!multiCellSelectable || !selectedRange) {
+      return false;
+    }
+
+    const { anchor, cursor } = selectedRange;
+
+    const startRow = Math.min(anchor.rowIdx, cursor.rowIdx);
+    const endRow = Math.max(anchor.rowIdx, cursor.rowIdx);
+    const startCol = Math.min(anchor.colIdx, cursor.colIdx);
+    const endCol = Math.max(anchor.colIdx, cursor.colIdx);
+
+    return rowIdx >= startRow && rowIdx <= endRow && colIdx >= startCol && colIdx <= endCol;
+  };
+
+  const isCellOnEdgeOfSelectedRange = (rowIdx: number, colIdx: number) => {
+    const cellOnEdge: TableDataCellProps['selectedOnEdge'] = { top: true, bottom: true, left: true, right: true };
+
+    if (!multiCellSelectable || !selectedRange) {
+      return cellOnEdge;
+    }
+
+    const { anchor, cursor } = selectedRange;
+
+    const startRow = Math.min(anchor.rowIdx, cursor.rowIdx);
+    const endRow = Math.max(anchor.rowIdx, cursor.rowIdx);
+    const startCol = Math.min(anchor.colIdx, cursor.colIdx);
+    const endCol = Math.max(anchor.colIdx, cursor.colIdx);
+
+    cellOnEdge.top = rowIdx === startRow && colIdx >= startCol && colIdx <= endCol;
+    cellOnEdge.bottom = rowIdx === endRow && colIdx >= startCol && colIdx <= endCol;
+    cellOnEdge.left = colIdx === startCol && rowIdx >= startRow && rowIdx <= endRow;
+    cellOnEdge.right = colIdx === endCol && rowIdx >= startRow && rowIdx <= endRow;
+
+    return cellOnEdge;
+  };
+
+  const tableRef = useRef<HTMLTableElement>(null);
+
+  // multiCellSelectable일 때, 테이블의 selectstart 이벤트를 막아서 텍스트 드래그 방지
+  useEffect(() => {
+    if (!tableRef.current) {
+      return;
+    }
+
+    const tableEl = tableRef.current;
+
+    const handleSelectStart = (event: Event) => {
+      if (multiCellSelectable) {
+        event.preventDefault();
+      }
+    };
+
+    tableEl.addEventListener('selectstart', handleSelectStart);
+
+    return () => {
+      tableEl.removeEventListener('selectstart', handleSelectStart);
+    };
+  }, [multiCellSelectable]);
+
+  const [isShiftKeyPressed, setIsShiftKeyPressed] = useState(false);
+
+  // Shift 키 누르고 있는 동안에만 range 선택 가능
+  useEffect(() => {
```

Comment:
이 부분에 isNative 분기를 해줘야 할 것 같습니다~ 
```suggestion
  useEffect(() => {
    if (isNative) return;
```